### PR TITLE
Build deckschrubber using FPM.

### PR DESCRIPTION
**Pull request by bzh-bzh**
Modeled after the marathon-deb pipeline.

Outputs a .changes file like this:
```
Format: 1.8
Date: Thu, 20 Dec 2018 14:10:54 -0800
Source: deckschrubber
Binary: deckschrubber
Architecture: amd64
Version: 0.6.0~ocf1
Distribution: stable
Urgency: medium
Maintainer: <@8b309ec4f0ea>
Description: Deckschrubber inspects images of a Docker Registry and removes those older than a given age.
Changes:
deckschrubber (0.6.0~ocf1) whatever; urgency=medium
 .
   * Package created with FPM.
Checksums-Sha1:
 08a203e25e1389df255ecd0270fd83f54f4fdc0a 2998632 deckschrubber_0.6.0~ocf1_amd64.deb
Checksums-Sha256:
 fa02d9bad7f9514887afc1587bc9679ae1d136432b1b185a699f528146c2cc58 2998632 deckschrubber_0.6.0~ocf1_amd64.deb
Files:
 6285b97c76b37ae8bfb6f4e80c420c29 2998632 default extra deckschrubber_0.6.0~ocf1_amd64.deb
```

### CODE REVIEW

File: Dockerfile
Comment (dkess) [inline Dockerfile]: Would it be possible to run this as the `nobody` user?

File: package.sh
Comment (dkess) [inline package.sh]: You can combine this into one by doing `git clone --branch $DECKSCHRUBBER_TAG ...`

**Review by dkess — APPROVED** [review]
Cool, lgtm

File: Dockerfile
Comment (kkuehlz) [inline Dockerfile:14]: Sometimes go projects want a `GOPATH` env variable too

File: Dockerfile
Comment (kkuehlz) [inline Dockerfile]: nit: would use `/opt` instead of `/tmp`

File: Jenkinsfile
Comment (kkuehlz) [inline Jenkinsfile]: nit: empty line

File: package.sh
Comment (kkuehlz) [inline package.sh:1]: Why does this need to be in a separate script? This logic can all go in the `Dockerfile`

File: Dockerfile
Comment (jvperrin) [inline Dockerfile]: You shouldn't need to do this in two steps, the stretch container has stretch-backports enabled already and `golang-1.10` is only available in backports as you mentioned, so it'll have to install it from there. I tested this by spinning up a new stretch container and just running the `apt-get install` with `golang-1.10` included in the packages above and that worked just fine.

That being said, this would have to be separate if it didn't have the version in the actual package name like this.

File: Dockerfile
Comment (jvperrin) [reply Dockerfile]: I think this is fine because the docker container is already run with `--user` so it should be running under a jenkins user already instead of root.

File: Jenkinsfile
Comment (jvperrin) [inline Jenkinsfile]: Instead of this, I'd try to use the [`packagePipeline`](https://github.com/ocf/shared-pipeline/blob/master/vars/packagePipeline.groovy) if you can to avoid duplication. It should be something like this, although the `Makefile` targets would need to be changed to just have `package_stretch` as the only target.

Then this whole file would just be this, although it would need some changes to the `Makefile` to achieve this:

```
packagePipeline(
    dists: ['stretch'],
)

// vim: ft=groovy
```

(I should go update marathon-deb too, I think it can be simplified a fair amount)

File: Makefile
Comment (jvperrin) [inline Makefile]: Instead of this, I'd just make the dist whatever you pass in to the build step, so something like `package_stretch` would set `stretch` as the dist to use. This also makes it easy to build for multiple dists if needed in the future.

File: package.sh
Comment (jvperrin) [inline package.sh:2]: Nice, thanks for setting this

File: package.sh
Comment (jvperrin) [inline package.sh:15]: I'd set the maintainer email in here too, since the default one in docker containers doesn't make that much sense. Either root@ocf or help@ocf are probably fine.

**Review by jvperrin — COMMENTED** [review]
Nice, thanks for packaging this! I've got a few comments, but otherwise looks good to me.

**Comment by bzh-bzh** [comment]
I believe it works with the shared pipeline now, but please confirm. .changes file now looks like this:
```
Format: 1.8
Date: Fri, 21 Dec 2018 14:41:25 -0800
Source: deckschrubber
Binary: deckschrubber
Architecture: amd64
Version: 0.6.0~ocf1
Distribution: stretch
Urgency: medium
Maintainer: root@ocf.berkeley.edu
Description: Deckschrubber inspects images of a Docker Registry and removes those older than a given age.
Changes:
deckschrubber (0.6.0~ocf1) whatever; urgency=medium
 .
   * Package created with FPM.
Checksums-Sha1:
 a0daa80abdf677dbafd76d6c93fd204cc92f8f88 2998620 deckschrubber_0.6.0~ocf1_amd64.deb
Checksums-Sha256:
 3f0c316cec689571e76004ef42e30c43964853c78af09b4dfaea8c70d558ae5b 2998620 deckschrubber_0.6.0~ocf1_amd64.deb
Files:
 0c72b4ea187a258fbf7b918f53ee6c15 2998620 default extra deckschrubber_0.6.0~ocf1_amd64.deb

```

**Review by kkuehlz — COMMENTED** [review]
sorry, i left comments yesterday but forgot the submit the review :(. lgtm, mostly nitpicks on my end

File: package.sh
Comment (jvperrin) [reply package.sh:1]: I do actually think this way is cleaner given the length of the script. If it were just like 3 lines then I'd say embedding it in the `Dockerfile` would be good, but as-is, I'd say a separate script is easier to follow, read, and modify.

Comment (bzh-bzh) [reply package.sh:1]: I'm not sure if I should make the directories for go and set the go env variables in the Dockerfile or the script (which is where it's set right now). Thoughts?

File: Dockerfile
Comment (bzh-bzh) [reply Dockerfile:14]: For now, that's set in package.sh

File: package.sh
Comment (kkuehlz) [reply package.sh:1]: I'll leave it to your discretion. What does and doesn't go in the script seems a bit arbitrary, which is why I suggested moving it all to the Dockerfile. I don't think it matters where you put it, but I would choose one and group `ENV PATH="${PATH}:/usr/lib/go-1.10/bin` with the other environment variables.

**Review by kkuehlz — APPROVED** [review]
Thanks!

**Comment by jvperrin** [comment]
I set up Jenkins to have a build step for this repo (just went to Jenkins, did "New Item," copied the config from the existing marathon-deb essentially, and then added @ocfbot to have access to the repo so it can check collaborators, and it looks like the PR build is running now.

File: package.sh
Comment (jvperrin) [inline package.sh:18]: I'd add in `--url https://github.com/fraunhoferfokus/deckschrubber` (or maybe even this repo, not sure which would be more useful there), but otherwise this looks good to me.

**Review by jvperrin — APPROVED** [review]
Nice, thanks for the great work on this!

It builds on Jenkins and the generated `.deb` looks fine to me (contains the right files and everything), although I added one small comment about the URL since it's currently just the default value which isn't really helpful.

File: package.sh
Comment (bzh-bzh) [reply package.sh:18]: Thanks for setting up the pipeline! I put this repo as the URL, since it links to deckschrubber's repo in the README too.

**Pull request merged**
